src/Compilers/Core/Portable/CodeGen/ReferenceDependencyWalker.cs

```diff
@@ -42,16 +44,16 @@ public static void VisitReference(Cci.IReference reference, EmitContext context)
 
         private static void VisitTypeReference(Cci.ITypeReference typeReference, EmitContext context)
         {
-            Debug.Assert(typeReference != null);
+            RoslynDebug.Assert(typeReference != null);
 
-            Cci.IArrayTypeReference arrayType = typeReference as Cci.IArrayTypeReference;
+            Cci.IArrayTypeReference? arrayType = typeReference as Cci.IArrayTypeReference;
             if (arrayType != null)
```

Comment:
```csharp
if (typeReference is CCI.IArrayTypeReference arrayType)
{
```
?

Reply:
I'm avoiding unnecessary refactorings in this pull request.